modules/assignment/src/main/java/com/intuit/wasabi/assignment/impl/AssignmentsImpl.java

```diff
@@ -497,12 +486,12 @@ public Assignment putAssignment(User.ID userID, Application.Name applicationName
                     }
 
                     // Add the assignment to the global list of userAssignments of the user
-                    if (assignment.getStatus() != Assignment.Status.EXPERIMENT_EXPIRED) {
+                    if (Experiment.State.PAUSED != experiment.getState()) { // the Experiment is set to PAUSED in the case of a reached endDate
```

Comment:
@bargenilesh @AndreaSuckro We'll need to make sure that we don't break the priority rules.